dace/sdfg/analysis/cutout.py

```diff
@@ -337,12 +365,16 @@ def multistate_cutout(cls,
         :param make_side_effects_global: If True, all transient data containers which are read inside the cutout but may
                                         be written to _before_ the cutout, or any data containers which are written to
                                         inside the cutout but may be read _after_ the cutout, are made global.
+        :param preserve_guids: If True, ensures that the GUIDs of graph elements contained in the cutout remain
+                               identical to the ones in their original graph. If False, new GUIDs will be generated.
+                               False by default - if make_copy is False, this has no effect by extension.
         :param override_start_block: If set, explicitly force a given control flow block to be the start block. If left
                                      None (default), the start block is automatically determined based on domination
                                      relationships in the original graph.
         :return: The created SDFGCutout or the original SDFG where no smaller cutout could be obtained.
         """
-        create_element = copy.deepcopy
+        def create_element(x: Union[ControlFlowBlock, InterstateEdge]) -> Union[ControlFlowBlock, InterstateEdge]:
+            return x.clone(keep_guid=preserve_guids)
```

Comment:
I would use an inline lambda, it's much cleaner

Reply:
Changed - note that this drops type information

Reply:
After removing the `clone` function I believe it is cleaner this way since this way provides type information, while lambdas completely remove the ability to type hint things (cleanly)